src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -40,15 +45,35 @@
   public void init() throws ServletException {
     super.init();
     setUserStore(UserStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+    setConversationStore(ConversationStore.getInstance());
   }
 
   /**
-   * Sets the UserStore used by this servlet. This function provides a common setup method for use
-   * by the test framework or the servlet's init() function.
+   * Sets the UserStore, MessageStore, and Conversation Store used by this servlet. This
+   * function provides a common setup method for use by the test framework or the servlet's
+   * init() function.
    */
   void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
+  void setMessageStore(MessageStore messageStore) { this.messageStore = messageStore; }
+  void setConversationStore(ConversationStore conversationStore) { this.conversationStore = conversationStore; }
+
+  /**
+   * Gets the correct stats from datastore and stores them in a map, simulating a json file
+   */
+  private void addStats(HashMap<String, String> map) {
+    // Retrieve sizes of each Datastore
+    Integer userSize = userStore.countTotalUsers();
+    Integer messageSize = messageStore.countTotalMessages();
+    Integer convSize = conversationStore.countTotalConversations();
```

Comment:
Verify whether conversationStore.countTotalConversations() is guaranteed to be non-null ?

Reply:
That method is guaranteed to never return a null, because what that method does is get the size of a list that is created in Conversation Store since the beginning, so if the list has no conversations it would return a 0 instead of null.